### PR TITLE
Remove unnecessary exceptions from analysis_options.yaml

### PR DESCRIPTION
**Pull request by goderbauer**
Inspired by Dan's comment on https://github.com/flutter/engine/pull/34986 I went through the other declared differences in analyzer options between the `flutter/flutter` and the `flutter/engine` repository. Turns out, some of these exceptions (namely `import_internal_library`, `prefer_final_fields`, and `sort_constructors_first`) are no longer necessary. The stated reason for why they are different from `flutter/flutter` don't appear to apply anymore.

This reduces the effective differences between `flutter/flutter` and `flutter/engine` down to three.

This PR also removes the special treatment of `todo`, `missing_required_param`, and `missing_return` as those are also no longer necessary as explained in https://github.com/flutter/flutter/pull/108747.

### CODE REVIEW

File: lib/ui/painting.dart [hunk 1]
```diff
@@ -6032,19 +6023,30 @@ class ImageDescriptor extends NativeFieldWrapperClass1 {
...
-    required int height,
-    int? rowBytes,
-    required PixelFormat pixelFormat,
-  }) {
+      ImmutableBuffer buffer, {
+        required int width,
+        required int height,
+        int? rowBytes,
+        required PixelFormat pixelFormat,
+      }) {
```
Comment (dnfield) [inline lib/ui/painting.dart:6031]: undo whitespace changes

**Review by dnfield — APPROVED** [review]
LGTM with nit

@yjbanov may want a chance to look at this

File: lib/ui/painting.dart (hunk 1 above)
Comment (goderbauer) [reply lib/ui/painting.dart]: Thanks for catching that. Fixed.

File: lib/web_ui/lib/src/engine/canvaskit/surface.dart [hunk 2]
```diff
@@ -26,14 +26,14 @@ typedef SubmitCallback = bool Function(SurfaceFrame, CkCanvas);
...
-
   SurfaceFrame(this.skiaSurface, this.submitCallback)
       : _submitted = false,
         assert(skiaSurface != null), // ignore: unnecessary_null_comparison
-        assert(submitCallback != null); // ignore: unnecessary_null_comparison
+        assert(submitCallback != null);
+
+  final CkSurface skiaSurface;
+  final SubmitCallback submitCallback;
+  final bool _submitted; // ignore: unnecessary_null_comparison
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/canvaskit/surface.dart:36]: What is this `// ignore: unnecessary_null_comparison` referring to?

File: lib/web_ui/lib/src/engine/canvaskit/surface.dart [hunk 3]
```diff
@@ -26,14 +26,14 @@ typedef SubmitCallback = bool Function(SurfaceFrame, CkCanvas);
...
 class SurfaceFrame {
-  final CkSurface skiaSurface;
-  final SubmitCallback submitCallback;
-  bool _submitted;
-
   SurfaceFrame(this.skiaSurface, this.submitCallback)
       : _submitted = false,
         assert(skiaSurface != null), // ignore: unnecessary_null_comparison
-        assert(submitCallback != null); // ignore: unnecessary_null_comparison
+        assert(submitCallback != null);
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/canvaskit/surface.dart:32]: Looks like `// ignore: unnecessary_null_comparison` is still needed here?

File: lib/web_ui/lib/src/engine/canvaskit/text.dart [hunk 4]
```diff
@@ -1044,6 +1039,10 @@ class _ParagraphCommand {
       _CkParagraphPlaceholder placeholderStyle)
       : this._(
             _ParagraphCommandType.addPlaceholder, null, null, placeholderStyle);
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/canvaskit/text.dart:1041]: Separate constructors from the fields by an empty line.

File: lib/web_ui/lib/src/engine/html/path/path.dart [hunk 5]
```diff
@@ -43,18 +64,14 @@ class SurfacePath implements ui.Path {
   ui.PathFillType _fillType = ui.PathFillType.nonZero;
   // Skia supports inverse winding as part of path fill type.
   // For Flutter inverse is always false.
-  bool _isInverseFillType = false;
+  final bool _isInverseFillType = false;
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/html/path/path.dart:67]: A final bool field seems like an unnecessary variable.

File: lib/web_ui/lib/src/engine/html/recording_canvas.dart [hunk 6]
```diff
@@ -51,7 +51,7 @@ class RawRecordingCanvas extends BitmapCanvas implements ui.PictureRecorder {
 
   late RecordingCanvas _canvas; // ignore: unused_field
 
-  bool _isRecording = true; // ignore: unused_field
+  final bool _isRecording = true; // ignore: unused_field
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/html/recording_canvas.dart:54]: Looks like `RawRecordingCanvas` class is used anywhere. I think it's dead code.

File: lib/web_ui/lib/src/engine/canvaskit/surface.dart (hunk 2 above)
Comment (goderbauer) [reply lib/web_ui/lib/src/engine/canvaskit/surface.dart]: Looks like `dart fix` messed up the comments. In general, `unnecessary_null_comparison` is ignored on a global level, so individual ignores are no longer needed. I will remove them here.

File: lib/web_ui/lib/src/engine/canvaskit/surface.dart (hunk 3 above)
Comment (goderbauer) [reply lib/web_ui/lib/src/engine/canvaskit/surface.dart:32]: Removed, see other comment.

File: lib/web_ui/lib/src/engine/canvaskit/text.dart (hunk 4 above)
Comment (goderbauer) [reply lib/web_ui/lib/src/engine/canvaskit/text.dart:1041]: Done.

File: lib/web_ui/lib/src/engine/html/path/path.dart (hunk 5 above)
Comment (goderbauer) [reply lib/web_ui/lib/src/engine/html/path/path.dart]: Removed and updated code that was using it.

File: lib/web_ui/lib/src/engine/html/recording_canvas.dart (hunk 6 above)
Comment (goderbauer) [reply lib/web_ui/lib/src/engine/html/recording_canvas.dart]: Deleted.

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**